crypto/rand/rand_lib.c

```diff
@@ -535,12 +535,26 @@ static EVP_RAND_CTX *rand_new_seed(OSSL_LIB_CTX *libctx)
     EVP_RAND *rand;
     RAND_GLOBAL *dgbl = rand_get_global(libctx);
     EVP_RAND_CTX *ctx;
+    const char *propq = dgbl->seed_propq;
     char *name;
 
     if (dgbl == NULL)
         return NULL;
-    name = dgbl->seed_name != NULL ? dgbl->seed_name : "SEED-SRC";
-    rand = EVP_RAND_fetch(libctx, name, dgbl->seed_propq);
+    if (dgbl->seed_name != NULL) {
+        name = dgbl->seed_name;
+    } else {
+        /*
+         * Default to our internal seed source.  This isn't part of the FIPS
+         * provider so we need to override any FIPS= properties.  There is
+         * no compelling reason to preserve any other parts of the property
+         * query since we know we need either the default or base providers
```

Comment:
How do we know this? Why can't a seed source come from some other provider?

Reply:
I did have some reasoning behind this but can't remember.  I'll update to merge property lists etc.